frontend/src/TourForm/TourForm.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  createStyles,
+  InputAdornment,
+  InputBase,
+  makeStyles,
+  Snackbar,
+  TextField,
+  Theme,
+} from "@material-ui/core";
+import axios from "axios";
+import React, { SyntheticEvent, useEffect, useState } from "react";
+import EditIcon from "@material-ui/icons/Edit";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import ActionButtons from "../ActionButtons/ActionButtons";
+import DragDrop from "DragDrop/DragDrop";
+import Alert from "@material-ui/lab/Alert";
+import { CREATE_MURAL_API, GET_ALL_TOUR } from "constants/constants";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    flexContainer: {
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "start",
+      width: "40vw",
+      maxWidth: "500px",
+      padding: theme.spacing(3),
+      paddingBottom: theme.spacing(0),
+    },
+    field: {
+      width: "100%",
+      margin: theme.spacing(0, 0, 2, 0),
+    },
+    title: {
+      fontSize: "180%",
+    },
+    muralCount: {
+      marginLeft: "260px",
+      fontSize: "large",
+    },
+  })
+);
+
+interface ITourFormProps {
+  handleCancel: () => void;
+}
+
+function TourForm({ handleCancel }: ITourFormProps) {
+  const [murals, setMurals] = useState<any>([]);
+  const [muralsInTour, setMuralsInTour] = useState<any>([]);
+  const [muralResults, setMuralResults] = useState<any>([]);
+  const [muralQuery, setMuralQuery] = useState<string>("");
+
+  const [description, setDescription] = useState<string>("");
+  const [editingDesc, setEditingDesc] = useState<boolean>(false);
+  const [hoveringDesc, setHoveringDesc] = useState<boolean>(false);
+
+  const [title, setTitle] = useState<string>("");
+  const [editingTitle, setEditingTitle] = useState<boolean>(false);
+  const [hoveringTitle, setHoveringTitle] = useState<boolean>(false);
+
+  const [popup, setPopup] = useState<boolean>(false);
+
+  const styles = useStyles();
+
+  const handleAddMural = (addedMural: any) => {
+    if (!addedMural) return;
```

Comment:
if `addedMural` is an object this might evaluate `{}` to true - is that okay?

Reply:
hopefully the code logic makes only valid murals come in as input